### PR TITLE
feat: added ContractCreator solidity example contract (#530)

### LINKED ISSUE #530: [Solidity Assembly Yul support] Contract Creator
### Problem

The Hedera network supports smart contracts as part of EVM equivalence.
Occasionally items have shown up that surfaced inequivalence at the language or opcode layer.

Solidity supports inline assembly usage of the Yul language.
However, it may not be clear what coverage the Hedera Smart Contract Service 2.0 has for them

### Solution

- Add solidity contract example that utilize contract creation options for assembly using Yul

- Add a contract under contracts/yul/contract-creator and tests under respective directories

Ref: https://docs.soliditylang.org/en/latest/yul.html#evm-dialect

### Alternatives

_No response_
```[tasklist]
### Tasks
- [x] create(v, p, n)
- [x] create2(v, p, n, s)
```

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- added ContractCreator soldiity example contract to cover 
  - [x] create(v, p, n)
  - [x] create2(v, p, n, s)
- added unit tests to cover ContractCreator contract

**Related issue(s)**:

Fixes #530 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/solidity/yul/contract-creator/ContractCreator.sol [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (Nana-EC) [inline contracts/solidity/yul/contract-creator/ContractCreator.sol:1]: Fix license

**Review by Nana-EC — COMMENTED** [review]
FIx license and extra extra line

File: contracts/solidity/yul/contract-creator/ContractCreator.sol (hunk 1 above)
Comment (quiet-node) [reply contracts/solidity/yul/contract-creator/ContractCreator.sol]: Done

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**